posthog/queries/session_recordings/test/__snapshots__/test_session_recording_list_from_session_replay.ambr

```diff
@@ -1,3 +1,698 @@
+  '
+  with events_session_ids AS
+    (SELECT distinct `$session_id` as session_id
+     FROM events PREWHERE team_id = 2
+     AND timestamp >= '2022-12-27 12:00:00'
+     AND timestamp <= '2023-01-04 12:00:00'
+     AND event IN ['custom-event']
+     AND ((event = 'custom-event'
+           AND (has(['test_action_filter-d1a7498b-4a40-4564-b64b-e20f010ba340'], "$session_id")
+                AND has(['345c7424-1b12-4917-9fb7-e9242514d85e'], "$window_id"))))
+     AND (has(['Firefox'], replaceRegexpAll(JSONExtractRaw(properties, '$browser'), '^"|"$', '')))
+     and notEmpty(session_id))
+  SELECT session_id,
+         any(team_id),
+         any(distinct_id),
+         min(min_first_timestamp) as start_time,
+         max(max_last_timestamp) as end_time,
+         dateDiff('SECOND', start_time, end_time) as duration,
+         argMinMerge(first_url) as first_url,
+         sum(click_count),
+         sum(keypress_count),
+         sum(mouse_activity_count),
+         round((sum(active_milliseconds)/1000)/duration, 2) as active_time
+  FROM session_replay_events PREWHERE team_id = 2
+  AND min_first_timestamp >= '2022-12-28 00:00:00'
+  AND max_last_timestamp <= '2023-01-04 00:00:00'
+  AND session_id in
+    (select session_id
+     from events_session_ids)
+  WHERE 1=1
+  GROUP BY session_id
+  HAVING 1=1
+  ORDER BY start_time DESC
+  LIMIT 51
+  OFFSET 0
+  '
+---
+# name: TestClickhouseSessionRecordingsListFromSessionReplay.test_all_filters_at_once
+  '
+  with distinct_ids_for_person as
+    (SELECT distinct_id,
+            argMax(person_id, version) as person_id
+     FROM person_distinct_id2 as pdi
+     INNER JOIN
+       (SELECT id
+        FROM person
+        WHERE team_id = 2
+        GROUP BY id
+        HAVING max(is_deleted) = 0) person ON person.id = pdi.person_id PREWHERE team_id = 2
+     GROUP BY distinct_id
+     HAVING argMax(is_deleted, version) = 0
+     and person_id = '01881f38-3ecf-0000-ec97-75960c8f9c65') ,
+       events_session_ids AS
+    (SELECT distinct `$session_id` as session_id
+     FROM events PREWHERE team_id = 2
+     AND timestamp >= '2023-05-04 12:00:00'
+     AND timestamp <= '2023-05-19 11:59:59'
+     AND event IN ['$pageview', 'custom-event']
+     AND ((event = 'custom-event'))
+     and notEmpty(session_id))
+  SELECT session_id,
+         any(team_id),
+         any(distinct_id),
+         min(min_first_timestamp) as start_time,
+         max(max_last_timestamp) as end_time,
+         dateDiff('SECOND', start_time, end_time) as duration,
+         argMinMerge(first_url) as first_url,
+         sum(click_count),
+         sum(keypress_count),
+         sum(mouse_activity_count),
+         round((sum(active_milliseconds)/1000)/duration, 2) as active_time
+  FROM session_replay_events PREWHERE team_id = 2
+  AND min_first_timestamp >= '2023-05-05 00:00:00'
+  AND max_last_timestamp <= '2023-05-18 23:59:59'
+  AND session_id in
```

Comment:
Hmm, shouldn't the session ID be a list of session IDs, why is this making another subquery? 🙈

Reply:
We have both... you can provide a fixed list of session ids... and/or when matching on events we check if the session_id was in the events query result